src/core/broker_config.cpp

```diff
@@ -17,37 +17,207 @@
+  const json::value_t type;
+  // A function to validate and store the value in the config's settings structure.
+  std::function<bool(const json&, BrokerConfig&)> validate_and_store;
+  // A function to store the default setting in the settings structure if the value is not present
+  // in the config file.
+  std::function<void(BrokerConfig&)> store_default;
+};
+
+// The CID must be a string representation of a UUID.
+bool ValidateAndStoreCid(const json& val, BrokerConfig& config)
+{
+  std::string str_val = val;
+  return etcpal_string_to_uuid(&config.settings.cid, str_val.c_str(), str_val.size());
+}
+
+// The UID takes the form:
+// "uid": {
+//   "type": < "static" | "dynamic" >,
+//   "manufacturer_id": <number, always present>,
+//   "device_id": <number, present only if type is "static">
+// }
+bool ValidateAndStoreUid(const json& val, BrokerConfig& config)
+{
+  if (val.contains("type") && val["type"].is_string() && val.contains("manufacturer_id") &&
+      val["manufacturer_id"].is_number_integer())
   {
-    "cid": "",
-    "uid_type": "dynamic",
-    "uid": "",
+    std::string type = val["type"];
+    int64_t manufacturer_id = val["manufacturer_id"];
 
-    "dns_sd": {
-      "service_instance_name": "",
-      "manufacturer": "ETC",
-      "model": "RDMnet Broker",
-      "scope": ""
-    },
+    if (manufacturer_id > 0 && manufacturer_id < 0x8000)
+    {
+      if (type == "static" && val.contains("device_id") && val["device_id"].is_number_integer())
+      {
+        int64_t device_id = val["device_id"];
+        if (device_id >= 0 && device_id <= 0xffffffff)
+        {
+          RdmUid static_uid;
+          static_uid.manu = static_cast<uint16_t>(manufacturer_id);
+          static_uid.id = static_cast<uint32_t>(device_id);
+          config.settings.SetStaticUid(static_uid);
+          return true;
+        }
+      }
+      else if (type == "dynamic" && !val.contains("device_id"))
+      {
+        config.settings.SetDynamicUid(static_cast<uint16_t>(manufacturer_id));
+        return true;
+      }
+    }
+  }
+  return false;
+}
 
-    "max_connections": 0,
-    "max_controllers": 0,
-    "max_controller_messages": 500,
-    "max_devices": 0,
-    "max_device_messages": 500,
-    "max_reject_connections": 1000
+// Store a generic string from the "dns_sd": {} object.
+template <size_t TruncationSize>
+bool ValidateAndStoreDnsString(const json& val, std::string& string)
+{
+  std::string str_val = val;
+  if (!str_val.empty())
+  {
+    string = str_val.substr(0, TruncationSize - 1);
+    return true;
   }
-)";
+  return false;
+}
 
-BrokerConfig::BrokerConfig()
+// Validate an arithmetic type and set it in the settings struct if it is within the valid range
+// for its type.
+template <class T>
+bool SetIfInRange(const json& val, T& setting)
 {
-  defaults_ = json::parse(kDefaultSettings);
+  int64_t int_val = val;
+  if (int_val >= std::numeric_limits<T>::min() && int_val <= std::numeric_limits<T>::max())
```

Comment:
!!  Nice!  But are we going to hit an embedded platform that doesn't support int64_t?

Reply:
Hmm. I haven't been considering embedded platforms to be valid targets for the broker... we should talk about this.

Reply:
Actually, that's a good point.  I was looking at the code with an eye towards re-use.